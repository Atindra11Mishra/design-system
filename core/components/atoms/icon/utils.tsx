import { FontVariationType } from './Icon';

const getOpticalSize = (size?: number) => {
  if (size && size <= 20) {
    return 20;
  } else if (size && size >= 48) {
    return 48;
  }

  return size;
};

const iconVariantMapper: Record<string, object> = {
  more_horiz: {
    weight: 600,
    opticalSize: 24,
  },
  delete: {
    fill: 0,
  },
  radio_button_checked: {
    fill: 0,
  },
  radio_button_unchecked: {
    fill: 0,
  },
  schedule: {
    fill: 0,
  },
  help_outline: {
    fill: 0,
  },
  check_circle_outline: {
    fill: 0,
  },
  highlight_of: {
    fill: 0,
  },
  credit_card: {
    fill: 0,
  },
  delete_outline: {
    fill: 0,
  },
  bookmark_border: {
    fill: 0,
  },
  work_outline: {
    fill: 0,
  },
  alarm: {
    fill: 0,
  },
  arrow_circle_up: {
    fill: 0,
  },
  copyright: {
    fill: 0,
  },
  query_builder: {
    fill: 0,
  },
  card_membership: {
    fill: 0,
  },
  arrow_circle_down: {
    fill: 0,
  },
  change_history: {
    fill: 0,
  },
  aspect_ratio: {
    fill: 0,
  },
  calendar_view_month: {
    fill: 0,
  },
  alarm_on: {
    fill: 0,
  },
  thumb_down_off_alt: {
    fill: 0,
  },
  important_devices: {
    fill: 0,
  },
  turned_in_not: {
    fill: 0,
  },
  add_circle_outline: {
    fill: 0,
  },
  speed: {
    fill: 0,
  },
  content_copy: {
    fill: 0,
  },
  timelapse: {
    fill: 0,
  },
};

export const getFontVariation = (iconName: string, variations?: FontVariationType, type?: string, size?: number) => {
  const defaultVariant = {
    fill: type === 'outlined' ? 0 : 1,
    weight: 400,
    grade: 0,
    opticalSize: getOpticalSize(size),
  };

  return { ...defaultVariant, ...iconVariantMapper[iconName], ...variations };
};
