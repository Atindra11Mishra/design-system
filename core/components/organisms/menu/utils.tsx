import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  menuTriggerRef?: any,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  subListRef?: any,
  isSubMenuTrigger?: boolean,
  triggerRef?: any,
  menuID?: string
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef);
      break;
    case 'Enter':
      handleEnterKey(focusedOption, menuTriggerRef);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      menuTriggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    case 'Tab':
      setOpenPopover?.(false);
      break;
    case 'ArrowRight':
      navigateSubMenu(isSubMenuTrigger, 'right', listRef, subListRef, triggerRef, menuID);
      break;
    case 'ArrowLeft':
      navigateSubMenu(isSubMenuTrigger, 'left', listRef, subListRef, triggerRef, menuID);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined, menuTriggerRef?: any) => {
  (focusedOption as HTMLElement)?.click();
  menuTriggerRef.current.focus();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];

  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption?.scrollIntoView?.({ block: 'center' });
};

const navigateSubMenu = (
  isSubMenuTrigger?: boolean,
  direction?: string,
  listRef?: any,
  subListRef?: any,
  triggerRef?: any,
  menuID?: string
) => {
  console.log('listRef', listRef, 'menuIDmenuIDmenuID', menuID, document.querySelector(`[data-name="${menuID}"]`));

  const element = document.querySelector(`[data-name="${menuID}"]`);
  const menuPlacement = element?.getAttribute('data-placement');
  console.log('dddddddplacement', menuPlacement, 'menuID', menuID, 'element', element);

  if (isSubMenuTrigger) {
    if (
      (direction === 'right' && menuPlacement?.includes('right')) ||
      (direction === 'left' && menuPlacement?.includes('left'))
    ) {
      const listItems = subListRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
      (listItems[0] as HTMLElement).focus();
    }
  } else if (
    (direction === 'left' && menuPlacement?.includes('right')) ||
    (direction === 'right' && menuPlacement?.includes('left'))
  ) {
    triggerRef?.current.focus();
  }
};
