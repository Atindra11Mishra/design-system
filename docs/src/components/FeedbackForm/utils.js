const positiveChoices = [
  {
    label: 'Easy to understand',
    name: 'positiveChoices',
    value: 'Easy to understand',
  },
  {
    label: 'Solved my use-case',
    name: 'positiveChoices',
    value: 'Solved my use-case',
  },
  {
    label: 'Easy to implement',
    name: 'positiveChoices',
    value: 'Easy to implement',
  },
  {
    label: 'Other',
    name: 'positiveChoices',
    value: 'Other',
  },
];

const negativeChoices = [
  {
    label: 'Too complex to understand',
    name: 'negativeChoices',
    value: 'Too complex to understand',
  },
  {
    label: 'Missing information',
    name: 'negativeChoices',
    value: 'Missing information',
  },
  {
    label: 'Did not solve my use-case',
    name: 'negativeChoices',
    value: 'Did not solve my use-case',
  },
  {
    label: 'Issue with sample code',
    name: 'negativeChoices',
    value: 'Issue with sample code',
  },
];

export const getFeedbackOptions = (selectedOption) => {
  switch (selectedOption) {
    case 'Yes':
      return {
        heading: 'What did you like?',
        choiceListOptions: positiveChoices,
        Feedback_Type_ID: '9282ac39-f587-4ae1-ab0a-0c3d23461b81',
        Selected_Option_ID: 'c6eaad04-8f97-4c7e-89df-73d530db1ce0',
        Comment_Box_ID: '1487fdc5-c078-4461-8506-a8ffcfde32c6',
      };

    case 'No':
      return {
        heading: 'Oh, what went wrong?',
        choiceListOptions: negativeChoices,
        Feedback_Type_ID: '9282ac39-f587-4ae1-ab0a-0c3d23461b81',
        Selected_Option_ID: '70aa236c-2889-436a-a359-0a97c654cb29',
        Comment_Box_ID: '13ed6dde-a0b1-4848-8dba-f36dfee0dc7c',
      };

    default:
      return;
  }
};
