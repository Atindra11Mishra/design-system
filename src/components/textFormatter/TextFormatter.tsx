import * as React from 'react';
import GenericText from '../_text';
import { BaseProps, BaseHtmlProps } from '@/utils/types';

export interface TextFormatterProps extends BaseProps, BaseHtmlProps<HTMLElement> {
  /**
   * Text to be rendered
   */
  children: React.ReactNode;
}

export const Bold = (props: TextFormatterProps) => {
  const { children, ...rest } = props;

  return (
    <GenericText data-test="DesignSystem-Bold" {...rest} componentType="b">
      {children}
    </GenericText>
  );
};

export const Italic = (props: TextFormatterProps) => {
  const { children, ...rest } = props;

  return (
    <GenericText data-test="DesignSystem-Italic" {...rest} componentType="i">
      {children}
    </GenericText>
  );
};

export const Underline = (props: TextFormatterProps) => {
  const { children, ...rest } = props;

  return (
    <GenericText data-test="DesignSystem-Underline" {...rest} componentType="u">
      {children}
    </GenericText>
  );
};

export const Strikethrough = (props: TextFormatterProps) => {
  const { children, ...rest } = props;

  return (
    <GenericText data-test="DesignSystem-Strikethrough" {...rest} componentType="s">
      {children}
    </GenericText>
  );
};
