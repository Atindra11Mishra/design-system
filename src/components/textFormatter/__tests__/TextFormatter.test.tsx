import * as React from 'react';
import { render } from '@testing-library/react';
import { Bold, Italic, Underline, Strikethrough, TextFormatterProps as Props } from '../TextFormatter';
import { testHelper, filterUndefined, testMessageHelper } from '@/utils/testHelper';

const children = 'TextFormatter text';

describe('Bold component', () => {
  describe('Snapshots', () => {
    const mapper = {};
    const testFunc = (props: Record<string, any>): void => {
      const attr = filterUndefined(props) as Props;

      it(testMessageHelper(attr), () => {
        const { baseElement } = render(<Bold>{children}</Bold>);
        expect(baseElement).toMatchSnapshot();
      });
    };

    testHelper(mapper, testFunc);
  });

  describe('Unit tests', () => {
    it('should render children', () => {
      const { getByTestId } = render(<Bold>{children}</Bold>);
      expect(getByTestId('DesignSystem-Bold')).toHaveTextContent(`${children}`);
    });

    it('should have bold tagname', () => {
      const { getByTestId } = render(<Bold>{children}</Bold>);
      expect(getByTestId('DesignSystem-Bold').tagName).toEqual('B');
    });

    it('should have font-weight=bold', () => {
      const { getByTestId } = render(<Bold>{children}</Bold>);
      expect(getByTestId('DesignSystem-Bold')).toHaveStyle('fontWeight: bold');
    });
  });
});

describe('Italic component', () => {
  describe('Snapshots', () => {
    const mapper = {};
    const testFunc = (props: Record<string, any>): void => {
      const attr = filterUndefined(props) as Props;

      it(testMessageHelper(attr), () => {
        const { baseElement } = render(<Italic>{children}</Italic>);
        expect(baseElement).toMatchSnapshot();
      });
    };

    testHelper(mapper, testFunc);
  });

  describe('Unit Tests', () => {
    it('should render children', () => {
      const { getByTestId } = render(<Italic>{children}</Italic>);
      expect(getByTestId('DesignSystem-Italic')).toHaveTextContent(`${children}`);
    });

    it('should have Italic tagname', () => {
      const { getByTestId } = render(<Italic>{children}</Italic>);
      expect(getByTestId('DesignSystem-Italic').tagName).toEqual('I');
    });

    it('should have font-weight=Italic', () => {
      const { getByTestId } = render(<Italic>{children}</Italic>);
      expect(getByTestId('DesignSystem-Italic')).toHaveStyle('fontStyle: italic');
    });
  });
});

describe('Strikethrough component', () => {
  describe('Snapshots', () => {
    const mapper = {};
    const testFunc = (props: Record<string, any>): void => {
      const attr = filterUndefined(props) as Props;

      it(testMessageHelper(attr), () => {
        const { baseElement } = render(<Strikethrough>{children}</Strikethrough>);
        expect(baseElement).toMatchSnapshot();
      });
    };

    testHelper(mapper, testFunc);
  });

  describe('Unit Tests', () => {
    it('should render children', () => {
      const { getByTestId } = render(<Strikethrough>{children}</Strikethrough>);
      expect(getByTestId('DesignSystem-Strikethrough')).toHaveTextContent(`${children}`);
    });

    it('should have Strikethrough tagname', () => {
      const { getByTestId } = render(<Strikethrough>{children}</Strikethrough>);
      expect(getByTestId('DesignSystem-Strikethrough').tagName).toEqual('S');
    });

    it('should have font-weight=Strikethrough', () => {
      const { getByTestId } = render(<Strikethrough>{children}</Strikethrough>);
      expect(getByTestId('DesignSystem-Strikethrough')).toHaveStyle('textDecoration: line-through');
    });
  });
});

describe('Underline component', () => {
  describe('Snapshots', () => {
    const mapper = {};
    const testFunc = (props: Record<string, any>): void => {
      const attr = filterUndefined(props) as Props;

      it(testMessageHelper(attr), () => {
        const { baseElement } = render(<Underline>{children}</Underline>);
        expect(baseElement).toMatchSnapshot();
      });
    };

    testHelper(mapper, testFunc);
  });

  describe('Unit tests', () => {
    it('should render children', () => {
      const { getByTestId } = render(<Underline>{children}</Underline>);
      expect(getByTestId('DesignSystem-Underline')).toHaveTextContent(`${children}`);
    });

    it('should have Underline tagname', () => {
      const { getByTestId } = render(<Underline>{children}</Underline>);
      expect(getByTestId('DesignSystem-Underline').tagName).toEqual('U');
    });

    it('should have font-weight=Underline', () => {
      const { getByTestId } = render(<Underline>{children}</Underline>);
      expect(getByTestId('DesignSystem-Underline')).toHaveStyle('textDecoration: underline');
    });
  });
});
