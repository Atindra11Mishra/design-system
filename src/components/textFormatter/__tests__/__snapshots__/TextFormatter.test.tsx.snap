// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bold component Snapshots 
  
 1`] = `
<body>
  <div>
    <b
      data-test="DesignSystem-Bold"
    >
      TextFormatter text
    </b>
  </div>
</body>
`;

exports[`Italic component Snapshots 
  
 1`] = `
<body>
  <div>
    <i
      data-test="DesignSystem-Italic"
    >
      TextFormatter text
    </i>
  </div>
</body>
`;

exports[`Strikethrough component Snapshots 
  
 1`] = `
<body>
  <div>
    <s
      data-test="DesignSystem-Strikethrough"
    >
      TextFormatter text
    </s>
  </div>
</body>
`;

exports[`Underline component Snapshots 
  
 1`] = `
<body>
  <div>
    <u
      data-test="DesignSystem-Underline"
    >
      TextFormatter text
    </u>
  </div>
</body>
`;
